### PR TITLE
Bump bdk version to alpha 9

### PR DESCRIPTION
**Pull request by thunderbiscuit**
This one took a bit of work. Just pushing my work to run it in the CI. Not done yet.

### CODE REVIEW

File: bdk-ffi/src/error.rs [hunk 1]
```diff
@@ -249,6 +255,21 @@ pub enum DescriptorError {
     Hex { e: String },
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum ExtractTxError {
+    #[error("an absurdly high fee rate of {fee_rate} sat/vbyte")]
+    AbsurdFeeRate { fee_rate: u64 },
```
Comment (reez) [inline bdk-ffi/src/error.rs:261]: I love `AbsurdFeeRate` 🧡

File: bdk-ffi/src/types.rs [hunk 2]
```diff
@@ -36,12 +39,29 @@ impl From<BdkCanonicalTx<'_, BdkTransaction, ConfirmationTimeHeightAnchor>> for
...
 
         CanonicalTx {
-            transaction: Arc::new(Transaction::from(tx.tx_node.tx)),
+            transaction: Arc::new(Transaction::from(tx.tx_node.tx.as_ref().clone())),
             chain_position,
         }
     }
 }
 
+// impl From<BdkCanonicalTx<'_, BdkTransaction, ConfirmationTimeHeightAnchor>> for CanonicalTx {
```
Comment (reez) [inline bdk-ffi/src/types.rs:48]: I think we can remove this commented out part here?

File: bdk-ffi/Cargo.toml [hunk 3]
```diff
@@ -20,6 +20,7 @@ default = ["uniffi/cli"]
 [dependencies]
 bdk = { version = "1.0.0-alpha.9", features = ["all-keys", "keys-bip39"] }
 bdk_esplora = { version = "0.11.0", default-features = false, features = ["std", "blocking"] }
+esplora-client = { version = "0.7.0", default-features = false, features = ["blocking-https-rustls"] }
```
Comment (reez) [inline bdk-ffi/Cargo.toml:23]: nice!

**Review by reez — APPROVED** [review]
ACK 90026505c70b5e4d2d805d2f82beb72538df1058

Just added one comment for something commented out to potentially remove.

File: bdk-ffi/src/types.rs (hunk 2 above)
Comment (thunderbiscuit) [reply bdk-ffi/src/types.rs:48]: Totally. My bad.

**Pull request merged**